### PR TITLE
Flattening ValueTuple test folder and update following mscorlib

### PR DESCRIPTION
**Pull request by jcouv**
This is updating the corefx tree with some changes I had to make while working on the mscorlib/desktop version of ValueTuple.
Also, I renamed the test files to flatten the test folder.

@stephentoub @tarekgh for review.

### CODE REVIEW

File: src/System.ValueTuple/tests/ExtensionsTests.cs [hunk 1]
```diff
@@ -5,13 +5,12 @@
 using System;
 using System.Text;
 using Xunit;
-using static ValueTupleTests;
 
 public class ExtensionsTests
```
Comment (stephentoub) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: As long as you're making changes, can you put this in a System.Tests namespace?

File: src/System.ValueTuple/tests/ExtensionsTests.cs [hunk 2]
```diff
@@ -673,27 +672,27 @@ private struct IntHolder
...
+            if (x12 > 0) { builder.Append(" " + x12); }
+            if (x13 > 0) { builder.Append(" " + x13); }
+            if (x14 > 0) { builder.Append(" " + x14); }
+            if (x15 > 0) { builder.Append(" " + x15); }
+            if (x16 > 0) { builder.Append(" " + x16); }
+            if (x17 > 0) { builder.Append(" " + x17); }
+            if (x18 > 0) { builder.Append(" " + x18); }
+            if (x19 > 0) { builder.Append(" " + x19); }
+            if (x20 > 0) { builder.Append(" " + x20); }
+            if (x21 > 0) { builder.Append(" " + x21); }
```
Comment (stephentoub) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: Why this change?  Are we not using a C# 6 compiler in all places you're compiling these tests?

**Comment by stephentoub** [comment]
LGTM

File: src/System.ValueTuple/tests/ValueTuple/UnitTests.cs [hunk 3]
```diff
@@ -29,7 +29,7 @@ private class ValueTupleTestDriver<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10>
         internal ValueTupleTestDriver(params object[] values)
         {
             if (values.Length > 10)
-                throw new ArgumentOutOfRangeException(nameof(values), "You must provide at most 10 values");
```
Comment (tarekgh) [inline src/System.ValueTuple/tests/ValueTuple/UnitTests.cs]: >(nameof(values), [](start = 53, length = 16)

Is this change because we need to have the test written the same way for the desktop which using older version of the compiler?

**Comment by tarekgh** [comment]
> public class TupleElementNamesAttributeTests

as @stephentoub pointed out, please added this under some namespace

---
Refers to: src/System.ValueTuple/tests/TupleElementNamesTests.cs:9 in 9de769b. [](commit_id = 9de769b2cda2c552737e19c5647dcf539ff938a4, deletion_comment = False)

**Comment by tarekgh** [comment]
LGTM too.

File: src/System.ValueTuple/tests/ExtensionsTests.cs (hunk 2 above)
Comment (jasonwilliams200OK) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: If this is going to be reverted, the changes in ValueTupleTests.cs seems related and can also be rolled back (except for the file rename/move).

Comment (jcouv) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: When building this code in the NetFxDev1 TFS branch, the string literals were rejected. I didn't check what version of the compiler is used there.
That's also why I had to remove the static import at the top.

File: src/System.ValueTuple/tests/ValueTuple/UnitTests.cs (hunk 3 above)
Comment (jcouv) [inline src/System.ValueTuple/tests/ValueTuple/UnitTests.cs]: Yes, all the changes above were needed for the code to compile in the NetFxDev1 branch.

File: src/System.ValueTuple/tests/ExtensionsTests.cs (hunk 1 above)
Comment (jcouv) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: Sure. Added `namespace System.ValueTuple.Tests` to all three test files.
I'll update my NetFxDev1 shelveset accordingly.

Comment (jcouv) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: Darn, I think you meant it should be `namespace System.Tests`. I'll fix that.

Comment (stephentoub) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: Yeah, basically having the tests in the same namespace + ".Tests" as the type being tested.

File: src/System.ValueTuple/tests/ExtensionsTests.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: Ok.  Makes sense.  What'd you think then about adding a langver restriction to the tests project so that misuse doesn't accidentally slip in?

Comment (jcouv) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: @stephentoub Good idea. 
The compiler used in NetFxDev1 is 4.0 (meaning C# 4). I'll added langver=4 to make syncing the code easier.

Comment (jcouv) [inline src/System.ValueTuple/tests/ExtensionsTests.cs]: So, I made a mistake above. NetDevFx1 builds mscorlib using `D:\NetFXDev1\src\Tools\x86\managed\v4.5\Roslyn\csc2.exe`, not version 4.0 that I thought.

From discussion with Aleksey, we think this corresponds to an unfinished version of C# 6, because this is a managed compiler (meaning Roslyn and C# 6), but `csc2` is a compiler server that never shipped. 
That explains why it's able to use some C# 6 features (expression-bodied properties), but not others (string interpolation with `$"..."`).
So I will have to remove the `langver` constraint.

**Comment by jcouv** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test  please

**Comment by jcouv** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test please

**Comment by jcouv** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test  please

**Comment by jcouv** [comment]
@dotnet-bot test  Innerloop OSX Release Build and Test please

**Comment by jcouv** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test  please

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please

**Comment by jcouv** [comment]
Test Innerloop OSX Debug Build and Test  please

**Comment by jcouv** [comment]
@stephentoub I apparently now have permission to merge to this repo, whereas previously you would do the merge for me. Can you confirm that's ok, and I can merge?

**Comment by stephentoub** [comment]
@jcouv, yup, you can go ahead :smile:
